clarify ado behavior (#6705)
dbt-labs/docs.getdbt.com: website/docs/docs/build/packages.md

#### Prerequisites

- To use native private packages, you must have one of the following Git providers configured in the **Integrations** section of your **Account settings**:
  - [GitHub](/docs/cloud/git/connect-github)
  - [Azure DevOps](/docs/cloud/git/connect-azure-devops)

>     - For Azure DevOps, use the `org/repo` path (not the `org_name/project_name/repo_name` path) with the project tier inherited from the integrated source repository.

Reviewer: I think the important thing to emphasize as a pre-req is that they only work in a single project within ADO. The specific syntax, I think you do a good job of covering below.

    - Private packages only work within a single Azure DevOps project. If your repositories are in different projects within the same organization, you can't reference them in the `private` key at this time.
  - Support for GitLab is coming soon.

#### Configuration

Use the `private` key in your `packages.yml` or `dependencies.yml` to clone package repos using your existing dbt Cloud Git integration without having to provision an access token or create a dbt Cloud environment variable. 

> The private key supports a two-layered-path: `org/repo`. dbt inherits the project path from your source repository.

Reviewer: I would just leave this out - the project is kind of a concept specific to ADO, so it's not exactly correct to say that that's the behavior for other providers.

> The private key supports a two-layered-path: `org/repo`. dbt inherits the project path from your source repository.

Author: ```suggestion
```

#### Configuration

Use the `private` key in your `packages.yml` or `dependencies.yml` to clone package repos using your existing dbt Cloud Git integration without having to provision an access token or create a dbt Cloud environment variable. 

> The private key supports a two path layer: `org/repo` and dbt inherites the project path from your source repository.

Author: ```suggestion
The private key supports a two-layered-path: `org/repo`. dbt inherits the project path from your source repository.
```

<File name="packages.yml">

```yaml
packages:
  - private: dbt-labs/awesome_repo # your-org/your-repo path
  - package: normal packages
  [...]
```

> :::tip

Author: maybe this shouldn't be a callout - can make this expandable or h4?

Normally ADO repos use the `org_name/project_name/repo_name` path. However, you can only use the `org/repo` path to configrue private packages. If the package repository in ADO is within the same project as your source repository, then specify `org/repo` in the `private` key. If the package repository _isn't_ within the same project as your source repository, then the private packages feature won't currently work:

<File name="packages.yml">

```yaml
packages:
  - private: my-org/my-repo # Works if your source repo and package repo are in the same project
```
</File>
:::

For Azure DevOps repositories, you can only specify the `org/repo` path. The project tier is inherited from the source repository currently integrated with dbt Cloud.

- If the package repository is within the same project as your source repository, you’re in the clear. Simply specify org/repo in the private block.
- If the package repository is not within the same project as your source repository, this feature will not work for you until a future update.

> - This currently only works within a single Azure DevOps project. If your repositories are in different projects within the same organization, you can't reference them in the `private` key at this time.

Reviewer: Could we add this as a "Prerequisite" of the feature? That, for ADO, the repo is located within the same org/project as your project's repo.

Reviewer 2: Agree with @ericbaumann here! We should move these bullets up to the "Prereqs" section.

Author: thanks @ericbaumann @runleonarun - updated it to this:

![Screenshot 2024-12-24 at 10 04 21](https://github.com/user-attachments/assets/e1688eb3-b3c0-4b0f-9616-3c54a8dc1901)

#### Configuration

Use the `private` key in your `packages.yml` or `dependencies.yml` to clone package repos using your existing dbt Cloud Git integration without having to provision an access token or create a dbt Cloud environment variable. 

The private key supports a two-layered-path: `org/repo`. dbt inherits the project path from your source repository.

<File name="packages.yml">

```yaml
packages:
  - private: dbt-labs/awesome_repo # your-org/your-repo path
  - package: normal packages
  [...]
```
</File>

> :::tip Azure DevOps repositories

Reviewer 2: Maybe this should be titled "Azure DevOps limitations?"

Author: that's a good flag! i usually try use more positive language and tend to use considerations more than limitations. would Azure DevOps considerations work?

Author: ![Screenshot 2024-12-24 at 10 04 24](https://github.com/user-attachments/assets/ff6f1149-c224-4784-a38d-b52ae302834f)

> Normally ADO repos use the `org_name/project_name/repo_name` path. However, private package supports only the `org/repo` path. If the ADO package repository is within the same project as your source repository, you can specify `org/repo` in the `private` key. If the package repository _isn't_ within the same project as your source repository, then the private packages feature won't currently work:

Reviewer 2: I recommend starting with the take away from  this callout. Maybe we can also simplify it a little?

```suggestion
Currently, the private package feature only works when the ADO package repository is within the same project as your source repository. Normally ADO repos use the `org_name/project_name/repo_name` path. However, private package supports only the `org/repo` path. 

You can use private packages by specifying `org/repo` in the `private` key:
```
